pkg/queue/cluster_queue_interface.go

```diff
@@ -102,16 +107,30 @@ type ClusterQueue interface {
 	Active() bool
 }
 
-var registry = map[kueue.QueueingStrategy]func(cq *kueue.ClusterQueue, wo workload.Ordering) (ClusterQueue, error){
-	kueue.StrictFIFO:     newClusterQueueStrictFIFO,
-	kueue.BestEffortFIFO: newClusterQueueBestEffortFIFO,
+func newClusterQueue(cq *kueue.ClusterQueue, wo workload.Ordering) (*clusterQueueBase, error) {
+	cqImpl := newClusterQueueImpl(queueOrderingFunc(wo), realClock)
+	err := cqImpl.Update(cq)
+	if err != nil {
+		return nil, err
+	}
+	return cqImpl, nil
 }
 
-func newClusterQueue(cq *kueue.ClusterQueue, wo workload.Ordering) (ClusterQueue, error) {
-	strategy := cq.Spec.QueueingStrategy
-	f, exist := registry[strategy]
-	if !exist {
-		return nil, fmt.Errorf("invalid QueueingStrategy %q", cq.Spec.QueueingStrategy)
+// queueOrderingFunc returns a function used by the clusterQueue heap algorithm
+// to sort workloads. The function sorts workloads based on their priority.
+// When priorities are equal, it uses the workload's creation or eviction
+// time.
+func queueOrderingFunc(wo workload.Ordering) func(a, b *workload.Info) bool {
```

Comment:
Nit:
I guess this function could be moved to the _impl file, and not be passed as a parameter.

Also maybe the interface is no longer necessary (in favor of a plain pointer to the implementation).

Reply:
> I guess this function could be moved to the _impl file, and not be passed as a parameter.

Makes sense, done.

> Also maybe the interface is no longer necessary (in favor of a plain pointer to the implementation).

No strong view here, let me try how this would look like, I will put in a separate commit.

Reply:
Done in https://github.com/kubernetes-sigs/kueue/pull/1934/commits/493cc34c88e94009b268dfe00447cb7d9a86ec9e